### ISSUE TITLE
Reaper is picking up nodes that have been removed from the ring

### ISSUE THREAD

**Issue by cswingler** [thelastpickle/cassandra-reaper#552]
**Spotted on 1.2.2 (0446d2d)**

Hey there,

It looks like Reaper is still reporting nodes that have been deleted from a ring as existing in both the UI and the API, though they're all marked as status `LEFT - DOWN`. Forgetting and re-adding the cluster (a Cassandra 2.0 cluster) does not fix the issue, so I imagine there's something not quite right about the discovery logic.

```
➜  ~   curl -uchrisswingler https://reaper/cluster/inboxcluster | jq '.nodes_status.endpointStates[].endpointNames'
Enter host password for user 'chrisswingler':
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7881  100  7881    0     0  51269      0 --:--:-- --:--:-- --:--:-- 51509
[
  "172.25.215.235",
  "172.25.213.204",
  "172.25.210.177",
  "172.25.210.111",
  "172.25.219.13",
  "172.25.213.88",
  "172.25.211.131",
  "172.25.210.7",
  "172.25.211.212",
  "172.25.211.179",
  "172.25.213.101",
  "172.25.213.185",
  "172.25.212.251",
  "172.25.209.251",
  "172.25.208.27",
  "172.25.214.221",
  "172.25.218.252",
  "172.25.212.72",
  "172.25.214.161",
  "172.25.216.213",
  "172.25.210.66",
  "172.25.217.36",
  "172.25.219.120",
  "172.25.215.160",
  "172.25.217.234",
  "172.25.219.162",
  "172.25.211.182",
  "172.25.219.146",
  "172.25.210.207",
  "172.25.217.230"
]
```

which is 30 matches, while `nodetool status` reports:

```
Datacenter: useast1-new
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address         Load       Tokens  Owns   Host ID                               Rack
UN  172.25.213.88   1.03 TB    256     4.9%   5f7e27f1-28e4-45c6-8fb5-e70253e02d0d  useast1b
UN  172.25.209.251  1.1 TB     256     5.1%   c7e189c6-a7cb-41ad-9e89-20ba7cec245b  useast1a
UN  172.25.208.27   1.13 TB    256     5.3%   cb2efd24-eba9-40fe-9380-18581a5a9ab9  useast1a
UN  172.25.218.252  1.17 TB    256     4.6%   90c0a5fa-fb48-4d38-a770-f5fafb21c982  useast1d
UN  172.25.211.179  1.22 TB    256     4.6%   c2725e74-f5b9-4a40-afa7-171434c0765b  useast1a
UN  172.25.211.212  1000.21 GB  256     4.8%   e647c31b-42fe-4a11-a50d-93146630ed1d  useast1a
UN  172.25.212.72   823.4 GB   256     4.9%   9c9ed8fa-155d-48d8-9bd4-0ad83871bfc2  useast1b
UN  172.25.215.235  988.18 GB  256     5.2%   b0513020-8f23-4ee4-b203-d9d5ff34da2f  useast1b
UN  172.25.217.234  1.33 TB    256     5.3%   7a7fb920-f300-41e6-b54a-75e12758e6c2  useast1d
UN  172.25.219.13   1.11 TB    256     5.0%   2705f111-987a-45bc-aacc-83c4e981ea34  useast1d
UN  172.25.213.204  1.73 TB    256     5.2%   375d4c56-919e-4db0-8c4b-e15658a284c9  useast1b
UN  172.25.210.207  1022.52 GB  256     4.9%   b915d1b9-26a5-47ff-bed0-3f00aac9eea1  useast1a
UN  172.25.214.161  1.04 TB    256     5.0%   1a675cf2-38e5-466b-81e2-ff80b7ea7db6  useast1b
UN  172.25.215.160  1017.02 GB  256     5.4%   c05f1f0d-55ec-4ea2-9b5d-9be178bac34d  useast1b
UN  172.25.210.66   989.99 GB  256     4.6%   e8ed0075-50d6-417c-9656-8dfc472c34b2  useast1a
UN  172.25.219.162  1.11 TB    256     4.5%   e30d20fe-a6ff-4da8-82b5-35da303985fa  useast1d
UN  172.25.213.101  1.1 TB     256     5.0%   a4034558-e285-4d3d-9269-57cc0f56ce52  useast1b
UN  172.25.217.36   1.18 TB    256     5.0%   e6db6b8e-f514-4670-80ed-3c016692da4d  useast1d
UN  172.25.210.7    941.77 GB  256     5.4%   ab59030b-b483-454e-a7da-3145dd102bd6  useast1a
UN  172.25.217.230  1.11 TB    256     5.1%   bfe81ea8-ab44-4762-b081-278140c76bdf  useast1d
```

which is only the 20 nodes that still exist.

Curiously, the `seed_hosts` column in `reaper_db.cluster` as well as the `seed_hosts` value reported by the API, both report only the 20 hosts that are online.

**Comment by cswingler** [comment]
Ah ha, that makes sense. I was a little used to Opscenter's behavior (it drops nodes from its UI as soon as they're marked as left). Thanks for the clarification!

**Comment by adejanovski** [comment]
I have an open PR with the fix.

**Comment by adejanovski** [comment]
Fixed in #582
